core/src/main/java/org/hisp/dhis/android/network/datavalue/DataValueDTO.kt

```diff
@@ -65,23 +65,21 @@ internal data class DataValueDTO(
             followup?.let { followUp(it) }
         }.build()
     }
+}
 
-    companion object {
-        fun fromDomain(dataValue: DataValue): DataValueDTO {
-            return DataValueDTO(
-                deleted = dataValue.deleted(),
-                dataElement = dataValue.dataElement()!!,
-                period = dataValue.period()!!,
-                orgUnit = dataValue.organisationUnit()!!,
-                categoryOptionCombo = dataValue.categoryOptionCombo()!!,
-                attributeOptionCombo = dataValue.attributeOptionCombo()!!,
-                value = dataValue.value(),
-                storedBy = dataValue.storedBy(),
-                created = dataValue.created()?.let { DateUtils.DATE_FORMAT.format(it) },
-                lastUpdated = dataValue.lastUpdated()?.let { DateUtils.DATE_FORMAT.format(it) },
-                comment = dataValue.comment(),
-                followup = dataValue.followUp(),
-            )
-        }
-    }
+internal fun DataValue.toDto(): DataValueDTO {
+    return DataValueDTO(
+        deleted = this.deleted(),
+        dataElement = this.dataElement()!!,
+        period = this.period()!!,
+        orgUnit = this.organisationUnit()!!,
+        categoryOptionCombo = this.categoryOptionCombo()!!,
+        attributeOptionCombo = this.attributeOptionCombo()!!,
+        value = this.value(),
+        storedBy = this.storedBy(),
+        created = this.created()?.let { DateUtils.DATE_FORMAT.format(it) },
```

Comment:
Should we use the date extensions created in this PR?

Reply:
Yes, but I didn't apply the extension to existing files (event those modified in this PR). We can do it as a separate task to move them all.